lib/get-tasks.js

```diff
@@ -1,27 +1,25 @@
 /**
  * Acceptable formats:
- * - #20jt35r
  * - #CORE-123
  */
 const getTasks = (source) => {
-    const ticketRefRegex = /(?<=\s|^)#(([A-Z]{2,10}-\d+)|(\w{7,10}))/g;
-    const ticketIdRegex = /#(?<taskId>([A-Z]{2,10}-\d+)|(\w{7,10}))/;
-    const customTicketIdRegex = /#[A-Z]{2,10}-\d+/;
-    const ids =
-      source.match(ticketRefRegex)?.map(v => ({
-        ...v.match(ticketIdRegex).groups,
-        isCustom: customTicketIdRegex.test(v),
-      })) || [];
+  const taskRefRegex = /(?<=\s|^)#[A-Z]{2,10}-\d+/g;
+  const refs = source.match(taskRefRegex) || [];
+  const uniqueRefs = new Set(refs);
 
-    return Object.values(
-      ids.reduce(
-        (tasks, task) => ({
-          ...tasks,
-          [task.taskId]: task,
-        }),
-        {}
-      )
-    );
-  };
+  let tasks = [];
+  for (const ref of uniqueRefs) {
+    tasks.push({
+      taskId: ref.slice(1), // remove # from the beginning of the reference
+    });
+  }
+  // sort tasks alphabetically
+  tasks.sort(function(refA, refB) {
+    if (refA.taskId < refB.taskId) return -1;
+    if (refA.taskId > refB.taskId) return 1;
+    return 0;
+  });
```

Comment:
is this needed? IMO we should keep tasks in order they appeared in the title

Reply:
Probably not, just my OCD wanted everything to be sorted 😅 I've now removed it to have a simpler logic 4fc47c6a790f3407443c43b56859ac028bb99d53.